sound/soc/sof/sof-audio.c

```diff
@@ -181,12 +189,34 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (++swidget->use_count > 1)
 		return 0;
 
+	/*
+	 * The scheduler widget for a pipeline is not part of the connected DAPM
+	 * widget list and it needs to be set up before the widgets in the pipeline
+	 * are set up. The use_count for the scheduler widget is incremented for every
+	 * widget in a given pipeline to ensure that it is freed only after the last
+	 * widget in the pipeline is freed. Skip setting up pipeline widget for static pipelines.
+	 */
+	if (swidget->id == snd_soc_dapm_scheduler || !swidget->dynamic_pipeline_widget)
+		goto set_up;
```

Comment:
I don't see any reference for ABI 3.19 in the patch itself which I was expecting based on the commit message.
The real deciding factor to do this is the dynamic property of a pipeline, which might well be depending on ABI 3.19?